manifests/init.pp

```diff
@@ -269,44 +274,45 @@
-  $user_hash          = {},
-  $configure_firewall = false,
-  $install_java       = $jenkins::params::install_java,
-  $repo_proxy         = undef,
-  $proxy_host         = undef,
-  $proxy_port         = undef,
-  $no_proxy_list      = undef,
-  $cli                = true,
-  $cli_ssh_keyfile    = undef,
-  $cli_tries          = $jenkins::params::cli_tries,
-  $cli_try_sleep      = $jenkins::params::cli_try_sleep,
-  $port               = $jenkins::params::port,
-  $libdir             = $jenkins::params::libdir,
-  $sysconfdir         = $jenkins::params::sysconfdir,
-  $manage_datadirs    = $jenkins::params::manage_datadirs,
-  $localstatedir      = $::jenkins::params::localstatedir,
-  $executors          = undef,
-  $slaveagentport     = undef,
-  $manage_user        = $::jenkins::params::manage_user,
-  $user               = $::jenkins::params::user,
-  $manage_group       = $::jenkins::params::manage_group,
-  $group              = $::jenkins::params::group,
-  $default_plugins    = $::jenkins::params::default_plugins,
-  $purge_plugins      = $::jenkins::params::purge_plugins,
+  $version              = $jenkins::params::version,
+  $lts                  = $jenkins::params::lts,
+  $repo                 = $jenkins::params::repo,
+  $package_name         = $jenkins::params::package_name,
+  $direct_download      = $::jenkins::params::direct_download,
+  $package_cache_dir    = $jenkins::params::package_cache_dir,
+  $package_provider     = $jenkins::params::package_provider,
+  $manage_service       = true,
+  $service_enable       = $jenkins::params::service_enable,
+  $service_ensure       = $jenkins::params::service_ensure,
+  $service_provider     = $jenkins::params::service_provider,
+  $config_hash          = {},
+  $plugin_hash          = {},
+  $job_hash             = {},
+  $user_hash            = {},
+  $configure_firewall   = false,
+  $install_java         = $jenkins::params::install_java,
+  $repo_proxy           = undef,
+  $proxy_host           = undef,
+  $proxy_port           = undef,
+  $no_proxy_list        = undef,
+  $cli                  = true,
+  $cli_ssh_keyfile      = undef,
+  $cli_tries            = $jenkins::params::cli_tries,
+  $cli_try_sleep        = $jenkins::params::cli_try_sleep,
+  $port                 = $jenkins::params::port,
+  $libdir               = $jenkins::params::libdir,
+  $sysconfdir           = $jenkins::params::sysconfdir,
+  $manage_datadirs      = $jenkins::params::manage_datadirs,
+  $localstatedir        = $::jenkins::params::localstatedir,
+  $executors            = undef,
+  $slaveagentport       = undef,
+  $manage_user          = $::jenkins::params::manage_user,
+  $user                 = $::jenkins::params::user,
+  $manage_group         = $::jenkins::params::manage_group,
+  $group                = $::jenkins::params::group,
+  $default_plugins      = $::jenkins::params::default_plugins,
+  $default_plugins_host = $::jenkins::params::default_plugins_host,
```

Comment:
I would really like to see a simple test in https://github.com/jenkinsci/puppet-jenkins/blob/master/spec/defines/jenkins_plugin_spec.rb that demonstrates changing this parameter changes the plugin download url.